backends/occa/ceed-occa.c

```diff
@@ -44,23 +44,24 @@ static int CeedError_Occa(Ceed ceed,
 // * CeedDestroy_Occa
 // *****************************************************************************
 static int CeedDestroy_Occa(Ceed ceed) {
-  const Ceed_Occa *impl=ceed->data;
-
+  int ierr;
+  Ceed_Occa *data=ceed->data;
   CeedDebug("\033[1m[CeedDestroy]");
-  occaDeviceFree(impl->device);
+  occaDeviceFree(*data->device);
+  ierr = CeedFree(&data->device); CeedChk(ierr);
+  ierr = CeedFree(&data); CeedChk(ierr);
   return 0;
 }
 
 // *****************************************************************************
 // * INIT
 // *****************************************************************************
 static int CeedInit_Occa(const char *resource, Ceed ceed) {
-  Ceed_Occa *impl;
-
+  int ierr;
+  Ceed_Occa *data;
   CeedDebug("\033[1m[CeedInit] resource='%s'", resource);
-  if (strcmp(resource, "/cpu/occa")
-      && strcmp(resource, "/omp/occa")
-      && strcmp(resource, "/gpu/occa"))
+  if (strcmp(resource, "/cpu/occa") && strcmp(resource, "/omp/occa") &&
+      strcmp(resource, "/ocl/occa") && strcmp(resource, "/gpu/occa"))
```

Comment:
I suspect we need to fix this naming sooner or later. OpenCL can drive a CPU or GPU. Our use of "gpu" seems to actually mean "CUDA device". We should probably call it that. (I recall some conversation months ago in which this came up.)